src/annotator/anchoring/pdf.js

```diff
@@ -155,38 +156,45 @@ export async function documentHasText() {
 /**
  * Return the text of a given PDF page.
  *
+ * The page text returned by this function should match the `textContent` of the
+ * text layer element that PDF.js creates for rendered pages. This allows
+ * offsets computed in the text to be reused as offsets within the text layer
+ * element's content. This is important to create correct Ranges for anchored
+ * selectors.
+ *
  * @param {number} pageIndex
  * @return {Promise<string>}
  */
-async function getPageTextContent(pageIndex) {
+function getPageTextContent(pageIndex) {
+  // If we already have or are fetching the text for this page, return the
+  // existing result.
   if (pageTextCache[pageIndex]) {
     return pageTextCache[pageIndex];
   }
 
-  // Join together PDF.js `TextItem`s representing pieces of text in a PDF page.
-  const joinItems = items => {
-    // Skip empty items since PDF.js leaves their text layer divs blank.
-    // Excluding them makes our measurements match the rendered text layer.
-    // Otherwise, the selectors we generate would not match this stored text.
-    // See the `appendText` method of `TextLayerBuilder` in PDF.js.
-    const nonEmpty = items
-      .filter(item => /\S/.test(item.str))
-      .map(item => item.str);
-    const textContent = nonEmpty.join('');
-    return textContent;
-  };
-
-  // Fetch the text content for a given page as a string.
-  const getTextContent = async pageIndex => {
+  const getPageText = async () => {
     const pageView = await getPageView(pageIndex);
     const textContent = await pageView.pdfPage.getTextContent({
       normalizeWhitespace: true,
     });
-    return joinItems(textContent.items);
-  };
+    let items = textContent.items;
+
+    // Older versions of PDF.js did not create elements in the text layer for
+    // text items that contained all-whitespace strings. Newer versions (after
```

Comment:
```suggestion
    // text items that contained all-whitespace strings. Versions >= v2.9.359
```

Reply:
I've made the above change to specify which version counts as "older" and "newer", so I won't repeat it here.